mitreattack/navlayers/generators/usage_generator.py

```diff
@@ -0,0 +1,132 @@
+from stix2 import Filter
+from itertools import chain
+from enum import Enum
+
+try:
+    from ..exporters.matrix_gen import MatrixGen
+    from ..core.exceptions import BadInput, typeChecker, categoryChecker
+    from ..core.layer import Layer
+    from ..generators.gen_helpers import remove_revoked_depreciated, get_attack_id
+except ValueError:
+    from mitreattack.navlayers.exporters.matrix_gen import MatrixGen
+    from mitreattack.navlayers.core.exceptions import BadInput, typeChecker, categoryChecker
+    from mitreattack.navlayers.core.layer import Layer
+    from mitreattack.navlayers.generators.gen_helpers import remove_revoked_depreciated, get_attack_id
+except ImportError:
+    from navlayers.exporters.matrix_gen import MatrixGen
+    from navlayers.core.exceptions import BadInput, typeChecker, categoryChecker
+    from navlayers.core.layer import Layer
+    from navlayers.generators.gen_helpers import remove_revoked_depreciated, get_attack_id
+
+
+class UnableToFindStixObject(Exception):
+    pass
+
+
+class UsageGenerator:
+    """Generates a Layer that shows techniques mapped to an input group, software or mitigation"""
+    def __init__(self, source, matrix='enterprise', local=None):
```

Comment:
Also here matrix should probably be domain.

Reply:
Got it.